src/System.Memory/src/System/Runtime/InteropServices/SequenceMarshal.cs

```diff
@@ -39,7 +39,14 @@ public static bool TryGetArray<T>(ReadOnlySequence<T> sequence, out ArraySegment
         /// </summary>
         public static bool TryGetReadOnlyMemory<T>(ReadOnlySequence<T> sequence, out ReadOnlyMemory<T> memory)
         {
-            return sequence.TryGetReadOnlyMemory(out memory);
+            if (!sequence.IsSingleSegment)
+            {
+                memory = default;
+                return false;
+            }
+
+            memory = sequence.First;
```

Comment:
Nice change! TryGetReadOnlyMemory and GetFirstBuffer were essentially identical.